projects/graphql-client/src/graphql-apollo-integration.test.ts

```diff
@@ -24,28 +24,40 @@ describe('GraphQl Apollo Integration Service', () => {
     ]
   });
 
-  const buildRequestString = (rootPath: string = 'test', params: string[] = []): string =>
+  const buildRequestString = (rootPath: string = 'test', params: string[] = [], fields: string[] = []): string =>
     new GraphQlRequestBuilder()
       .withSelects({
-        path: rootPath,
-        children: [{ path: 'id' }, { path: 'value' }],
-        ...params.map(param => ({ path: param }))
+        path: `${rootPath}${params.length > 0 ? `(${params.map(param => `${param}: ${param}`).join(',')})` : ''}`,
+        children: [{ path: 'id' }, { path: 'value' }, ...fields.map(field => ({ path: field }))]
       })
       .build();
 
-  const buildServerResponse = (rootPath: string = 'test', params: string[] = []) => ({
+  const buildRequestIdString = (rootPath: string = 'test', fields: string[] = []): string =>
+    buildRequestString(rootPath, ['id'], fields);
+
+  const buildServerResponse = (rootPath: string = 'test', fields: string[] = []) => ({
     data: {
       [`${rootPath}`]: {
         id: 'foo',
         value: 'bar',
-        ...params
-          .map(param => ({ [`${param}`]: param }))
-          .reduce((previousVal, current) => ({ ...previousVal, ...current }), {}),
+        ...fields.map(field => ({ [`${field}`]: field })),
```

Comment:
This appears to be spreading an array into an object.

Reply:
oh hmm.. I want them to be added as properties. I need to reduce it into an object before spreading